agent/api/transitiondependency_test.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package api
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestUnmarshalOldTransitionDependencySet(t *testing.T) {
+	bytes := []byte(`{
+	  "ContainerDependencies": [
+	    {
+	      "ContainerName": "container",
+	      "SatisfiedStatus": "RUNNING",
+	      "DependentStatus": "RUNNING"
+	    }
+	  ]
+	}`)
+	unmarshalledTdMap := TransitionDependenciesMap{}
+	err := json.Unmarshal(bytes, &unmarshalledTdMap)
+	assert.NoError(t, err)
+	assert.Len(t, unmarshalledTdMap, 1)
+	assert.NotNil(t, unmarshalledTdMap[ContainerRunning])
+	dep := unmarshalledTdMap[ContainerRunning].ContainerDependencies
+	assert.Len(t, dep, 1)
+	assert.Equal(t, dep[0].ContainerName, "container")
+	assert.Equal(t, dep[0].SatisfiedStatus, ContainerRunning)
+	assert.Equal(t, dep[0].DependentStatus, ContainerStatusNone)
+}
+
+func TestUnmarshalNewTransitionDependencySet(t *testing.T) {
+	bytes := []byte(`{"1":{"ContainerDependencies":[{"ContainerName":"container","SatisfiedStatus":"RUNNING"}]}}`)
```

Comment:
Can you confirm this will not be `PULLED` instead of `1`?

Reply:
The marshal keeps the `ContainerStatus` as it is, and this was `"1"` instead of `"PULLED"`

Reply:
https://play.golang.org/p/Qg4gemS1oup. Not sure about why golang wouldn't use the custom marshal for the object when it is the map's key

Reply:
I am not sure if that's the case, because that field in struct is still int(not a pointer) and hits the custom marshaler. This is only when the field is the key in a map.

Reply:
I think this is because golang will use the `TextMarshaler` for encoding the key of the map, so the key object will need to implement the `TextMarshaler` to customize the name. I think it's fine to have an integer here.

Reply:
Won't this cause an issue when a new container state is introduced in the future?